ext/posix/posix.c

```diff
@@ -784,7 +788,11 @@ PHP_FUNCTION(posix_getgrnam)
 #if defined(ZTS) && defined(HAVE_GETGRNAM_R) && defined(_SC_GETGR_R_SIZE_MAX)
 	buflen = sysconf(_SC_GETGR_R_SIZE_MAX);
 	if (buflen < 1) {
+#if defined(__FreeBSD__) && defined(_SC_PAGESIZE)
+		buflen = sysconf(_SC_PAGESIZE);
+#else
 		RETURN_FALSE;
+#endif
```

Comment:
So I have already asked about this and probably not in a clear way. So my main question was don't you just do this:

```c
#if defined(__FreeBSD__)
	buflen = sysconf(_SC_PAGESIZE);
#else
	buflen = sysconf(_SC_GETGR_R_SIZE_MAX);
#endif
	if (buflen < 1) {	
		RETURN_FALSE;
```

The only reason that I could see for not doing this is that you somehow expect that this is going to be supported on FreeBSD in the future. Is that the case? If so, I'm not sure we should be predicting what might happen (especially if it's more likely not going to happen). But if you really think that it's quite likely, then I'm fine with it.

In such case you should still check the return value of of the second sysconf call even though it is unlikely to fail. So something like this should be done:
```c
	buflen = sysconf(_SC_GETPW_R_SIZE_MAX);
	if (buflen < 1) {
#if defined(__FreeBSD__)
		buflen = sysconf(_SC_PAGESIZE);
		if (buflen < 1) {
			RETURN_FALSE;
		}
#else
		RETURN_FALSE;
#endif
```